### ISSUE TITLE
[Bug] False positive "variable is assigned but not used" in case of extended variable assignment

### ISSUE THREAD

**Issue by JaPyR** [MarketSquare/robotframework-robocop#971]
### What happened?

![image](https://github.com/MarketSquare/robotframework-robocop/assets/7773301/c1d23c9e-e90e-4481-acbd-0b2436db69eb)

MRE:
```robotframework
*** Settings ***
Library     DemoLib.py


*** Test Cases ***
DemoCase
    ${item}    Prepare Item
    Log    ${item.x}


*** Keywords ***
Prepare Item
    ${item}    Get Item
    ${item.x}    Set Variable    abc
    RETURN    ${item}
```

```python
from dataclasses import dataclass

@dataclass
class Item:
    x: str = ""

def get_item():
    return Item()
```

### What command/code did you try to run?

`robocop -i 0920 Demo.robot`

### What is the full error message?

 `[I] 0920 Variable '${item.x}' is assigned but not used (unused-variable)`

### What did you expect to happen instead?

Assigning to object attributes should not be marked as unused variable

### Robocop version

4.1.1

**Comment by JaPyR** [comment]
Very similar case is when object is passed to keyword and only object attributes are referenced inside. It results in warning "Keyword argument is not used", but not sure whether it deserve for separate bug.

**Comment by bhirsz** [comment]
It's one of the reason why I put the variable rules for so long - attribute handling or dynamic resolution of variables with static tool like Robocop is difficult. Like you noticed, when we're setting the variable we consider whole name as variable name. I could introduce a change that will try to recognize attribute access in a bit smarter way. 

As for second situation, Robocop should be able to detect such cases already but there could be additional bug. I will check and see if I can fix it also.

**Comment by bhirsz** [comment]
Also, "setting attribute" is also kind of "using the variable" so it should not be reported even in this case:

```
Update Item
    [Arguments]    ${item}
    ${item.x}    Set Variable    updated value
```

**Comment by bhirsz** [comment]
I've actually noticed something similar to your second case in #944

**Comment by bhirsz** [comment]
I have fairy easy workaround for now - we're listing unused-variable at the end of the keyword/test case. I can check if reported variable contains possible attribute access and if yes, then check if there is matching 'base' in declared variables.

**Comment by bhirsz** [comment]
Released in 4.2.0
